### PR TITLE
Incorrect volume/image displayed for listing

### PR DESCRIPTION
**Pull request by alexander-astrand**

### CODE REVIEW

File: frontend/marketplace/src/components/AuctionResultsCard.vue [hunk 1]
```diff
@@ -35,7 +36,11 @@ interface CardDetailsList {
 
 const props = defineProps<AuctionResultsCardProps>();
 const denom = ref("");
+const volume = computed(() => { 
+  return Number(props.cardData.creditCollection.retiredAmount) + Number(props.cardData.creditCollection.activeAmount)
+ })
```
Comment (jschill) [inline frontend/marketplace/src/components/AuctionResultsCard.vue:41]: Just to confirm. These amounts are never floating numbers, right?

Try this in your console: `Number(1.1)+Number(1.3)`

File: frontend/marketplace/src/components/AuctionCard.vue [hunk 2]
```diff
@@ -26,15 +31,20 @@ const handleViewDetailsClick = () => {
...
-        convertIPFStoHTTPS(
-          auctionData?.creditCollection?.creditData?.nodes[0].mediaFiles
-            ?.nodes[0].url,
-        ) || auctionCard
-      "
-    />
+    <div v-if="showSpinner">
+      <CustomSpinner :visible="showSpinner" />
+    </div>
+    <div v-if="showSpinner === false">
```
Comment (jschill) [inline frontend/marketplace/src/components/AuctionCard.vue:37]: `!showSpinner` to be consistent with the if above or `v-else` intead of `v-if`?

File: frontend/marketplace/src/utils/query-builder.ts [hunk 3]
```diff
@@ -194,6 +194,8 @@ export class ListingsQueryBuilder {
                     pricePerCreditDenom
                     creditCollection{
                         creditType
+                        retiredAmount
+                        activeAmount
```
Comment (jschill) [inline frontend/marketplace/src/utils/query-builder.ts:198]: 👍

File: frontend/marketplace/src/components/ImageGallery.vue [hunk 4]
```diff
@@ -2,32 +2,41 @@
...
   () => props.imageArray,
   (newValue) => {
     activeImageURL.value = newValue[0];
+    showSpinner.value = false;
   },
 );
 
 const handleActiveImage = (url: string) => {
   activeImageURL.value = url;
+  showSpinner.value = false;
```
Comment (jschill) [inline frontend/marketplace/src/components/ImageGallery.vue:25]: Why is this needed?

**Review by jschill — APPROVED** [review]

File: frontend/marketplace/src/components/AuctionResultsCard.vue (hunk 1 above)
Comment (alexander-astrand) [reply frontend/marketplace/src/components/AuctionResultsCard.vue:41]: I do get 2.4000000000000004 when doing that.. I see that parseInt is used in AuctionDetail.vue instead, is that a better solution?

File: frontend/marketplace/src/components/AuctionCard.vue (hunk 2 above)
Comment (alexander-astrand) [reply frontend/marketplace/src/components/AuctionCard.vue]: Fixed!

File: frontend/marketplace/src/components/ImageGallery.vue (hunk 4 above)
Comment (alexander-astrand) [reply frontend/marketplace/src/components/ImageGallery.vue:25]: It's not, fixed!

File: frontend/marketplace/src/components/AuctionResultsCard.vue (hunk 1 above)
Comment (jschill) [reply frontend/marketplace/src/components/AuctionResultsCard.vue]: @alexander-astrand just wanted to point out that if you are working with decimals/floats in javascript you need to use Decimal.js (or similar library) or convert the value to an int (by multiplying) before you do the math

Comment (alexander-astrand) [reply frontend/marketplace/src/components/AuctionResultsCard.vue]: Thanks for pointing that out, I'll have a look at that! Do you reckon parseInt does the trick here, or would another solution be better? Not really any need for decimals here as I gathered it, and based on the component already using parseInt!
